### ISSUE TITLE
Parses invalid class code

### ISSUE THREAD

**Issue by ljharb** [typescript-eslint/typescript-eslint#4072]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```tsx
        class Foo extends React.Component {
          contructor(props) {
            super(props);
            this.initialValues = {
              test: '',
            };
          }

          render = () => {
            return (
              <Component
                initialValues={this.props.initialValues || this.initialValues}
              >
                {formikProps => (
                  <Input {...formikProps} />
                )}
              </Component>
            );
          }
        }
```

**Expected Result**

A parse error, because `super()` is not valid except in `constructor() {}` and this is `contructor() {}`

**Actual Result**

No parse error.

**Additional Info**

Confusion over this led me to file https://github.com/eslint/eslint/issues/15234, but eslint 8 is actually handling this correctly because of the typo.

Related: https://github.com/babel/babel/issues/13903

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `5.2.0` |
| `TypeScript`                | `3.9.10` |
| `ESLint`                    | `8.1.0` |
| `node`                      | `17.0.1` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/1852

We are built on top of TS's parser, which is designed to be "forgiving" so it can handle incomplete code in usecases like an IDE.

We would like to add strict enforcement that when we convert the AST to ESTree it is done so it strictly adheres to the spec, but we haven't had the bandwidth for that yet.
